crates/fuel-core/src/schema/tx/types.rs

```diff
@@ -620,6 +624,71 @@ impl Transaction {
     }
 }
 
+#[derive(Union, Debug)]
+pub enum DryRunTransactionStatus {
+    Success(DryRunSuccessStatus),
+    Failed(DryRunFailureStatus),
+}
+
+impl DryRunTransactionStatus {
+    pub fn new(tx_status: TransactionExecutionResult) -> Self {
+        match tx_status {
+            TransactionExecutionResult::Success { result } => {
+                DryRunTransactionStatus::Success(DryRunSuccessStatus { result })
+            }
+            TransactionExecutionResult::Failed { result, reason } => {
+                DryRunTransactionStatus::Failed(DryRunFailureStatus { result, reason })
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct DryRunSuccessStatus {
+    result: Option<VmProgramState>,
+}
+
+#[Object]
+impl DryRunSuccessStatus {
+    async fn program_state(&self) -> Option<ProgramState> {
+        self.result.map(Into::into)
+    }
+}
+
+#[derive(Debug)]
+pub struct DryRunFailureStatus {
+    result: Option<VmProgramState>,
+    reason: String,
+}
+
+#[Object]
+impl DryRunFailureStatus {
+    async fn program_state(&self) -> Option<ProgramState> {
+        self.result.map(Into::into)
+    }
+
+    async fn reason(&self) -> String {
+        self.reason.clone()
+    }
+}
+
+pub struct DryRunTransaction(pub TransactionExecutionStatus);
```

Comment:
I'm not a big fan of this name, in reality it's still the status not the tx.

Also, should this be `pub`?

Reply:
We use the inner `pub` [here](https://github.com/FuelLabs/fuel-core/pull/1631/files#diff-2507047aabf2de10d20852af1dd31abceed668d66964340ce6fd0143d93c8a46R263)

I agree with your first point.  What alternative do you like `DryRunStatus` `DryRunTransactionStatus` `DryRunTransactionExecutionStatus`

Reply:
I forgot I already used `DryRunTransactionStatus` for an enum.  It's a bit verbose but maybe `DryRunTransactionExecutionStatus` is best?

Reply:
I prefer verbose over unclear :+1: